app/javascript/components/cloud-volume-form/cloud-volume-form.schema.js

```diff
@@ -0,0 +1,46 @@
+import { componentTypes, validatorTypes } from '@@ddf';
+
+const createSchema = (fields, promise, edit, loadSchema) => ({
+  fields: [
+    {
+      component: componentTypes.SELECT,
+      name: 'ext_management_system_id',
+      id: 'ext_management_system_id',
+      label: __('Storage Manager'),
+      placeholder: __('<Choose>'),
+      onChange: value => API.options(`/api/cloud_volumes?ems_id=${value}`).then(loadSchema()),
+      loadOptions: () => promise.then(({ resources }) => resources.map(({ id, name }) => ({ value: id, label: __(name) }))),
+      isDisabled: edit,
+      isRequired: true,
+      validate: [{ type: validatorTypes.REQUIRED }],
+    },
+    {
+      component: componentTypes.TEXT_FIELD,
+      name: 'name',
+      id: 'name',
+      label: __('Volume Name'),
+      isRequired: true,
+      validate: [{ type: validatorTypes.REQUIRED }],
+    },
+    {
+      component: componentTypes.TEXT_FIELD,
+      name: 'size',
+      id: 'size',
+      label: __('Size (in bytes)'),
+      type: 'number',
+      isRequired: true,
+      step: 1024 * 1024 * 1024,
+      validate: [{ type: validatorTypes.REQUIRED }],
+    },
+    {
+        component: componentTypes.TEXT_FIELD,
+        name: 'edit',
+        id: 'edit',
+        type: 'hidden',
+        value: edit,
+    },
+    ...fields,
```

Comment:
The `volume_type` field that's coming from this `fields` needs an additional [`resolveProps`](https://data-driven-forms.org/schema/resolve-props#heading-resolveprops#resolveprops) function which we cannot set on the API side as its value is a JS function.

```suggestion
    ...fields.slice(0, idx-1),
    {
      fields[idx],
      resolveProps: ({ options }, _input, { getState }) => ({
        options: options.filter(({ condition }) => !condition || parseCondition(condition, getState().values)),
      }),
    },
    ...fields.slice(idx),
```

The `idx` containes the index of the field we are enhancing and the `slice` helps us to get the fields before and after. The `resolveProps` function goes through all the options in the dropdown and uses the condition (if set) to filter out some values. There are some [special cases](https://github.com/ManageIQ/manageiq-providers-amazon/pull/657/files#diff-89fd56220c0624c072f49be1ec3b9b06153377a98664a770ed5723413b18216eR157-R174) when we should not show some of the options and the DDF didn't expose any other way to do this :confused: 

I am not expecting this to work for the first try, we might need to do a few iterations on this together...